components/core/src/clp/streaming_archive/MetadataDB.hpp

```diff
@@ -75,6 +75,7 @@ class MetadataDB {
                 epochtime_t begin_timestamp,
                 epochtime_t end_timestamp,
                 std::string const& file_path,
+                std::string const& file_id,
```

Comment:
1. Naming here is confusing. Still, we are using "file" to refer different things. We should be more explicit here, how about: `file_path` -> `original_path`
2. I might be wrong, but whenever a file_id is given, do we still need `original_path`? `file_id` should be unique iiuc. Maybe we should make two APIs to ensure either `original_path` or `file_id` is given but not both.

Reply:
1. Maybe we just go with `file_split_id` as the method parameter?
2. True. Technically `segment_id` wouldn't be necessary either. Not sure what's the easiest way to achieve this though. Might require a more significant code change.